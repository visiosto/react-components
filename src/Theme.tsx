// Copyright (c) 2021 Visiosto oy
// Licensed under the Apache License, Version 2.0

import React from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

type AnyIfEmpty<T extends object> = keyof T extends never ? any : T;

export type ThemeProps = { children: React.ReactNode; theme: AnyIfEmpty<DefaultTheme> };

function Theme({ children, theme }: ThemeProps) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

export default Theme;
