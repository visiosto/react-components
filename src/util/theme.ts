// Copyright (c) 2021 Visiosto oy
// Licensed under the Apache License, Version 2.0

import './DefaultTheme';

import { DefaultTheme } from 'styled-components';

const fonts = {
  heading: 'utopia-std, Times, "Times New Roman", serif',
  main: 'source-sans-pro, "Helvetica Neue", Helvetica, Arial, sans-serif',
  code: 'source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace',
};

// The maximum sizes of different viewports.
const sizes = {
  mobileS: '20rem', // 320px
  mobileM: '23.4375rem', // 375px
  mobileL: '26.5625rem', // 425px
  tablet: '48rem', // 768px
  laptop: '64rem', // 1024px
  laptopL: '90rem', // 1440px
  fourK: '160rem', // 2560px
};

const devices = {
  mobileS: `min-width: ${sizes.mobileS}`,
  mobileM: `min-width: ${sizes.mobileM}`,
  mobileL: `min-width: ${sizes.mobileL}`,
  tablet: `min-width: ${sizes.tablet}`,
  laptop: `min-width: ${sizes.laptop}`,
  laptopL: `min-width: ${sizes.laptopL}`,
  fourK: `min-width: ${sizes.fourK}`,
};

const layout = {
  marginMobile: '2rem',
  marginTablet: '4rem',
  marginDesktop: '16rem',
  marginTitleMobile: '0.5rem',
  marginTitleTablet: '1rem',
};

const borders = {
  commonRadius: '25px',
  cardRadius: '0.5rem',
};

const theme: DefaultTheme = {
  fonts: { ...fonts },
  sizes: { ...sizes },
  devices: { ...devices },
  layout: { ...layout },
  borders: { ...borders },
};

export default theme;
