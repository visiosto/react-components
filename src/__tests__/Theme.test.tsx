// Copyright (c) 2021 Visiosto oy
// Licensed under the MIT License

import React from 'react';
import { render } from '@testing-library/react';

import Theme from '../Theme';

import theme from '../util/theme';

describe('Theme component', () => {
  it('renders correctly', () => {
    const { container, getByText } = render(<Theme theme={theme}>Content</Theme>);

    expect(container).toMatchSnapshot();

    expect(getByText('Content')).toBeInTheDocument();
  });
});
