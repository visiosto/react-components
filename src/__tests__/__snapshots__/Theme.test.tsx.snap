// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme component renders correctly 1`] = `
<div>
  Content
</div>
`;
